core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -85,6 +85,9 @@ export class ExplorationPlayerStateService {
     private statsReportingService: StatsReportingService,
     private urlService: UrlService
   ) {
+    this.init();
```

Comment:
Why is this required?

Reply:
Ah never mind, I think should be fine to do this. Thanks!

Reply:
Since the constructor will be automatically called in unit tests, it is hard to test or spy on the constructor. So, we have created a function to manually trigger and tests different edge cases.